src/azure-cli-core/azure/cli/core/credential_helper.py

```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import re
+import json
+from enum import Enum
+
+
+class CredentialType(Enum):
+    STORAGE_SAS_TOKEN = ('([?&;])sig=[\w%-]+', r'\1sig=_REDACTED_SAS_TOKEN_SIG_', 'Storage SAS token')
+    CODE = ('([?&;])code=[\w%+/=-]+', r'\1code=_REDACTED_CODE_', 'Several types of keys/secrets are passed with a query parameter "code"')
+    AZCOPY_TOKEN = ('([?&/]|%2F)(SourceKey|DestKey)(=|%3A|%3D|:)(?:[\w/-]|%2F)+(?:==|%3D%3D)', r'\1\2\3_REDACTED_AZCOPY_TOKEN_', 'AzCopy token')
+    JWT_TOKEN = ('(?:eyJ0eXAi|eyJhbGci)[\w\-.~+/%]*', '_REDACTED_JWT_TOKEN_', 'JWT token')
+    BEARER_TOKEN = ('("|\'|%22)(bearer |bearer%20)[\w\-.~+/]{100,}("|\'|%22)', r'\1\2_REDACTED_BEARER_TOKEN_\3', 'Bearer token')
+    SSH_KEY = ('("|\')(ssh-rsa )AAAA[\w\-.~+/]{100,}("|\')', r'\1\2_REDACTED_SSH_KEY_\3', 'SSH key')
+    EMAIL_ADDRESS = ('\b[\w.%#+-]+(%40|@)([a-z0-9.-]*\.[a-z]{2,})\b', r'_REDACTED_EMAIL_\1\2', 'Email address')
+    GUID = ('([0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12})\b', '_REDACTED_GUID_', 'GUID')
```

Comment:
It is a good practice to always provide explanations and examples for regular expressions, as regular expressions can be difficult to understand for other code readers.

Reply:
Totally agree so I add test cases for each type

Reply:
also added comments as explanation